src/chess/net/net_trivial.rs

```diff
@@ -0,0 +1,440 @@
+use crate::chess::chess_game::{ChessGame, ChessPosition, BOARD_SIZE};
+use crate::game::common::{GamePosition, IGame};
+use crate::game::mcts::ValueFunction;
+use crate::game::net::{self, TwoHeadedNetBase};
+use once_cell::sync::Lazy;
+
+pub struct TrivialNet {}
+impl TrivialNet {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl ValueFunction<ChessGame> for TrivialNet {
+    fn evaluate(
+        &mut self,
+        position: &ChessPosition,
+    ) -> (f32, Vec<(<ChessGame as IGame>::Move, f32)>) {
+        let (position, is_flipped) = net::flip_pos_if_needed(*position);
+        let positionf = position.get_flip();
+
+        let b = position.get_raw_board();
+        let bf = positionf.get_raw_board();
+
+        let white = b.color_combined(chess::Color::White);
+        let pawns_w = b.pieces(chess::Piece::Pawn) & white;
+        let knights_w = b.pieces(chess::Piece::Knight) & white;
+        let bishops_w = b.pieces(chess::Piece::Bishop) & white;
+        let rooks_w = b.pieces(chess::Piece::Rook) & white;
+        let queens_w = b.pieces(chess::Piece::Queen) & white;
+        let kings_w = b.pieces(chess::Piece::King) & white;
+
+        let black = bf.color_combined(chess::Color::White);
+        let pawns_b = bf.pieces(chess::Piece::Pawn) & black;
+        let knights_b = bf.pieces(chess::Piece::Knight) & black;
+        let bishops_b = bf.pieces(chess::Piece::Bishop) & black;
+        let rooks_b = bf.pieces(chess::Piece::Rook) & black;
+        let queens_b = bf.pieces(chess::Piece::Queen) & black;
+        let kings_b = bf.pieces(chess::Piece::King) & black;
+
+        let mut val = 0.0;
+        val += dot_product(pawns_w, PAWNS);
+        val -= dot_product(pawns_b, PAWNS);
+        val += dot_product(knights_w, KNIGHTS);
+        val -= dot_product(knights_b, KNIGHTS);
+        val += dot_product(bishops_w, BISHOPS);
+        val -= dot_product(bishops_b, BISHOPS);
+        val += dot_product(rooks_w, ROOKS);
+        val -= dot_product(rooks_b, ROOKS);
+        val += dot_product(queens_w, QUEENS);
+        val -= dot_product(queens_b, QUEENS);
+        let endgame = (queens_w == chess::EMPTY
+            || (bishops_w == chess::EMPTY && (knights_w | bishops_w).popcnt() <= 1))
+            && (queens_b == chess::EMPTY
+                || (rooks_b == chess::EMPTY && (knights_b | bishops_b).popcnt() <= 1));
+        if endgame {
+            val += dot_product(kings_w, KINGS_ENDGAME);
+            val -= dot_product(kings_b, KINGS_ENDGAME);
+        } else {
+            val += dot_product(kings_w, KINGS);
+            val -= dot_product(kings_b, KINGS);
+        }
+        val = 2.0 / (1.0 + (val * -10.0).exp()) - 1.0;
+
+        let moves_probs = TwoHeadedNetBase::calc_moves_probs(position.get_legal_moves(), &POLICY);

```

Comment:
each move has a constant value right? e.g moving a piece from d2 to d4 is always some value V?
do you have an intuition why is this better than random?

Reply:
moving a knight to the center is generally better than moving it to the side, moving a king to the corner is generally better than moving it to the center, ect. Also i think the differences are not huge, except the zeros i added in the end of table which are knights promotions.
I assume this is only slightly better than uniform distribution

Reply:
cool